cli/src/semgrep/run_scan.py

```diff
@@ -422,7 +422,12 @@ def run_scan(
         metrics.add_project_url(project_url)
         metrics.add_integration_name(environ.get("SEMGREP_INTEGRATION_NAME"))
         metrics.add_configs(configs)
-        metrics.add_engine_type(engine_type)
+        metrics.add_engine_config(
+            engine_type,
+            ("all" if allow_untrusted_validators else "pro_rules")
+            if run_secrets and not disable_secrets_validation
+            else None,
```

Comment:
Optional: nested conditional expressions, especially the python way, make my brain explode,.But this specific occurrence is manageable. Others might also be ok, so feel free to ignore and resolve.

Reply:
This is definitely close to the limit, imo, but I think this is more understandable than introducing an if statement into this block.